test/DGmethods/compressible_Navier_Stokes/MWE_external_topography.jl

```diff
@@ -0,0 +1,267 @@
+using MPI
+using CLIMA
+using CLIMA.Mesh.Topologies
+using CLIMA.Mesh.Topography
+using CLIMA.Mesh.Grids
+using CLIMA.DGmethods
+using CLIMA.DGmethods.NumericalFluxes
+using CLIMA.MPIStateArrays
+using CLIMA.LowStorageRungeKuttaMethod
+using CLIMA.ODESolvers
+using CLIMA.GenericCallbacks
+using CLIMA.Atmos
+using CLIMA.VariableTemplates
+using LinearAlgebra
+using StaticArrays
+using Logging, Printf, Dates
+using CLIMA.Vtk
+using DelimitedFiles
+using Dierckx
+
+@static if haspkg("CuArrays")
+    using CUDAdrv
+    using CUDAnative
+    using CuArrays
+    CuArrays.allowscalar(false)
+    const ArrayType = CuArray
+else
+    const ArrayType = Array
+end
+
+if !@isdefined integration_testing
+    const integration_testing =
+        parse(Bool, lowercase(get(ENV,"JULIA_CLIMA_INTEGRATION_TESTING","false")))
+end
+
+
+Npoly = 4
+
+###################################################################
+# USER DEFINED REGION NAME based ON THE GRID FILE TO BE DOWNLOADED
+###################################################################
+region_name        = "monterey"
+topography_db_type = "NOAA_XYZ"
+
+
+
+##################################################################
+# READING AND INTERPOLATING EXTERNAL GRID
+##################################################################
+if topography_db_type == "NOAA_XYZ"
+    region_file_name_hdr = string(region_name, ".hdr");
+    region_file_name_xyz = string(region_name, ".xyz");
+    TEST_DIR             = @__DIR__
+    topo_files_str       = string(TEST_DIR, "/TopographyFiles/")
+    if !isfile(topo_files_str)
+        #create directory IF directory not created yet:
+        mkpath(topo_files_str)    
+    end
+    
+    #NOAA HEADER FILE
+    header_file_name_and_path = string(topo_files_str, region_file_name_hdr);
+    if !isfile(header_file_name_and_path)
+        url_path = string("https://web.njit.edu/~smarras/TopographyFiles/NOAA/", region_file_name_hdr);
```

Comment:
Can we do something besides your personal website here?

Reply:
Any server would work as long as we know it won't be closed.

Reply:
I guess this is largely a bigger project question for how to deal with "data" files. @simonbyrne might have thoughts.

Reply:
Can we download them directly from somewhere else? Or are they being preprocessed?

A decent permalink option might be figshare.com? If you have a tarball, I can upload it and set up DataDeps.jl (which handles local caching).

Reply:
The central DB in NOAA is here: https://www.ngdc.noaa.gov/mgg/global/global.html
however, the specific subregions need to be extracted from a global file. This can be done either manually and then download only the region of interest (as they do in WRF) or --which I haven't implemented yet-- automatically extract the region from the global file. The global file is 377MB large.

@simonbyrne